src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3125,8 +3125,27 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                                       ASR::down_cast<ASR::List_t>(type)->m_type, nullptr);
                 return false;
             } else if (ASR::is_a<ASR::Tuple_t>(*type)) {
+                int i = 0;
                 index = ASRUtils::EXPR(tmp);
-                int i = ASR::down_cast<ASR::IntegerConstant_t>(ASRUtils::EXPR(tmp))->m_n;
+                ASR::expr_t* val = ASRUtils::expr_value(index);
+                if (!val) {
+                    throw SemanticError("Runtime Indexing with " + ASRUtils::type_to_str_python(type) +
```

Comment:
Wait. Add an error test for this in https://github.com/lcompilers/lpython/tree/main/tests/errors and register in https://github.com/lcompilers/lpython/blob/main/tests/tests.toml.